translation/src/main/java/com/google/cloud/trace/zipkin/translation/TraceTranslator.java

```diff
@@ -89,15 +89,13 @@ private void finishTrace(Trace.Builder traceBuilder, Collection<Trace> converted
     }
   }
 
-  private String convertTraceId(long zipkinTraceId) {
-    // Stackdriver trace ID's are 128 bits = 16 bytes * 8
-    ByteBuffer idBuffer = ByteBuffer.allocate(16);
-    idBuffer.putLong(zipkinTraceId);
-    idBuffer.putLong(zipkinTraceId);
-    StringBuilder idBuilder = new StringBuilder();
-    for (byte b : idBuffer.array()) {
-      idBuilder.append(String.format("%02x", b));
-    }
-    return idBuilder.toString();
+  // Stackdriver trace ID's are 128 bits = 16 bytes * 8
+  private String convertTraceId(Span zipkinSpan) {
+    // zipkin conditionally encodes its id as 16 chars if the high bits are zero
+    // This unconditionally encodes as 32 chars as Stackdriver trace IDs are fixed length.
+    char[] result = new char[32];
+    Util.writeHexLong(result, 0, zipkinSpan.traceIdHigh);
```

Comment:
No, it's not an OSS tool. However, it is a pretty small use case, and I'm not sure if there are many projects that will have a mix of Stackdriver and Zipkin traces. The mismatch with the logging context may be a bigger issue if you think Zipkin users reference that frequently. We can pad with 0's if you think that would be better. Anyway, this problem won't exist once Zipkin trace ID's are 128-bit. :)